### DISCUSSION TITLE
Nextjs 13

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [mantinedev/mantine#2829] [Q&A] [answered]
Hi. I don't know how to configure mantine in nextjs 13. please help. thanks.

**Comment by rtivital** [comment, marked as answer]
If you want to use app directory, see https://github.com/mantinedev/mantine-next-template/tree/next-13-app
Otherwise, you can use template the same way as with Next 12 with pages directory.

**Reply by mikkurogue** [reply]
Adding to this, it seems that all components must have the 'use client' notation that use Mantine components. So defining a layout might be tricky as it on my end causes the layout rendering on the server only to allow client components to work, so using a Container component for instance would have to be put in to a wrapper and given the 'use client' notation.

I assume this will be fixed in iterations of Next13 (and emotion) to allow for context hooks to trigger on the server (in some fashion)

**Comment by MadadiSaeed** [comment]
Thank you so much.
